solver_spec.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'solver'
+describe Solver do
+  let(:solver) { Solver.new }
+
+  describe '#factorial' do
+    it 'returns 1 when given 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'returns 1 when given 1' do
+      expect(solver.factorial(1)).to eq(1)
+    end
+
```

Comment:
- [ ] Kindly consider adding a test case when the given negative value which raises an exception